rclcpp/src/rclcpp/node_interfaces/node_graph.cpp

```diff
@@ -134,6 +134,50 @@ NodeGraph::get_service_names_and_types() const
   return services_and_types;
 }
 
+std::map<std::string, std::vector<std::string>>
+NodeGraph::get_service_names_and_types_by_node(
+  const std::string & node_name,
+  const std::string & namespace_) const
+{
+  rcl_names_and_types_t service_names_and_types = rcl_get_zero_initialized_names_and_types();
+  rcl_allocator_t allocator = rcl_get_default_allocator();
+  rcl_ret_t ret = rcl_get_service_names_and_types_by_node(
+    node_base_->get_rcl_node_handle(),
+    &allocator,
+    node_name.c_str(),
+    namespace_.c_str(),
+    &service_names_and_types);
+  if (ret != RCL_RET_OK) {
+    auto error_msg = std::string("failed to get service names and types by node: ") +
+      rcl_get_error_string().str;
+    rcl_reset_error();
+    if (rcl_names_and_types_fini(&service_names_and_types) != RCL_RET_OK) {
+      error_msg +=
+        std::string(", failed also to cleanup service names and types, leaking memory: ") +
+        rcl_get_error_string().str;
+    }
+    throw std::runtime_error(error_msg + rcl_get_error_string().str);
```

Comment:
```suggestion
        rcl_get_error_string().str;
        rcl_reset_error();
    }
    throw std::runtime_error(error_msg);
```

Right?

Reply:
I natively copied the error handling from the `get_service_names_and_types()` function above. Should I fix those too in this PR?

Reply:
Oh, well up to you, but I'd say you should at least fix this one, since it's more than just a style thing.

Reply:
For sure. Are you fine if I fix the other case in this PR too or should I create a follow up?

Reply:
Fixing both is fine by me. I can review it if you do either way.

Reply:
Done in 029f7eaef775cbfb3d34031f7252d6c6462a9039.